src/App.tsx

```diff
@@ -1,93 +1,230 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import {
+  CSSTransition,
+  TransitionGroup,
+} from 'react-transition-group';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/Todolist/Todolist';
+import { Error } from './components/Error/Error';
+import { Todo } from './types/Todo';
+import { TodoTitle } from './types/TodoTitle';
+import { TodoCompleted } from './types/TodoCompleted';
+
+enum FilteredStatus {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
+  const user = useContext(AuthContext);
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [filterTodos, setFilterTodos] = useState<string>(FilteredStatus.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [changeTodos, setChangeTodos] = useState(true);
+  const [loader, setLoader] = useState<number>(0);
+  const [loaderDeleating, setLoaderDeleating] = useState<number[]>([]);
+  const [loaderAllTodos, setLoaderAllTodos] = useState<number[]>([]);
+
+  const showError = (text: string) => {
+    setErrorMessage(text);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 2000);
+  };
+
+  const addTodo = async () => {
+    if (newTodoTitle.trim().length === 0) {
+      showError('Title can\'t be empty');
+
+      return;
+    }
+
+    try {
+      const newTodo = await createTodo(user?.id, newTodoTitle);
+
+      setLoader(newTodo.id);
```

Comment:
```suggestion
      setLoader(newTodo.id);

      const newTodo = await createTodo(user?.id, newTodoTitle);
```
став перед запитом, щоб він крутив поки іде запит на бек

Reply:
Свариться лінтер 

<img width="1052" alt="Знімок екрана 2022-12-26 о 18 06 42" src="https://user-images.githubusercontent.com/105112047/209566349-e7192fb7-2cbe-4dd0-ae7d-0f1fb49a29e9.png">